application/client/src/app/ui/views/toolbar/chart/zoomer/state.ts

```diff
@@ -0,0 +1,230 @@
+        this._parent
+            .env()
+            .subscriber.register(
+                this._parent.ilc().channel.ui.sidebar.resize(() => this._resize()._update()),
+            );
+        this._fetch(this._session.stream.len()).catch((err: Error) => {
+            this._logger.error(err);
+        });
+    }
+
+    public isLoading(): boolean {
+        return this._loading;
+    }
+
+    public noData(): boolean {
+        let displayFilter: boolean = false;
+        if (
+            this._filters &&
+            this._filters.data.datasets &&
+            this._filters.data.datasets.length > 0
+        ) {
+            displayFilter = true;
+        }
+        return !this._loading && !displayFilter;
+    }
+
+    private _update() {
+        if (this._filters !== undefined && this._filters.width > 0) {
+            this._fetch(this._filters.width).catch((err: Error) => {
+                this._logger.error(err);
+            });
+        }
+    }
+
+    private _changeColors(entities: StoredEntity<FilterRequest>[]) {
+        if (!this._filters) {
+            return;
+        }
+        entities.forEach((entity: StoredEntity<FilterRequest>) => {
+            this._filters !== undefined &&
+                this._filters.data.datasets.forEach((dataset: any) => {
+                    if (
+                        dataset.label === entity.definition.uuid &&
+                        dataset.backgroundColor !== entity.definition.colors.background
+                    ) {
+                        dataset.backgroundColor = entity.definition.colors.background;
+                    }
+                });
+        });
+        this._filters.update();
+    }
+
+    private _resize(): State {
+        this._parent.detectChanges();
+        this._width = this._vcRef.element.nativeElement.clientWidth;
+        return this;
+    }
+
+    private _updateCursor() {
+        const prev: boolean = this.isCursorVisible;
+        const streamSize: number = this._session.stream.len();
+        if (streamSize > 0 && this._width < streamSize) {
+            this.isCursorVisible = true;
+        } else {
+            this.isCursorVisible = false;
+        }
+        if (this.isCursorVisible !== prev) {
+            if (this.isCursorVisible === false) {
+                this._service.setPosition({
+                    session: this._session.uuid(),
+                    position: { full: 0, left: 0, width: 0 },
+                });
+            }
+            this._parent.detectChanges();
+        }
+    }
+
+    private _draw() {
+        this._loading = true;
+        this._extractData();
+        if (this._filters === undefined) {
+            this._createChart();
+        } else {
+            this._filters.data = {
+                datasets: this._datasets,
+                labels: new Array(this._labelCount).fill(''),
+            };
+            this._filters.update();
+        }
+        this._loading = false;
+        this._parent.detectChanges();
```

Comment:
Not clear how flag `loading` can "update" something as soon as here no any async code. Maybe I'm wrong :/

Reply:
No no, you're right. I wanted to move the `this._loading = true` into the async method  `_fetch(width: number)` where we update our data. Must've missed it when I was resolving the comments